lib/Sema/TypeCheckPattern.cpp

```diff
@@ -1045,7 +1060,13 @@ NullablePtr<Pattern> TypeChecker::trySimplifyExprPattern(ExprPattern *EP,
         return nullptr;
     }
   }
-  return nullptr;
+
+  const auto *BLE = dyn_cast_or_null<BooleanLiteralExpr>(
+      subExpr->getSemanticsProvidingExpr());
```

Comment:
Tiny nit: `getSemanticsProvidingExpr` never returns null, so `dyn_cast` should be sufficient here